libmscore/fret.h

```diff
@@ -21,43 +21,145 @@ class StringData;
 class Chord;
 class Harmony;
 
+// Keep this in order - used for comparisons
+enum class FretDotType : signed char {
+      NORMAL,
+      CROSS,
+      SQUARE,
+      TRIANGLE
+      };
+
+
+enum class FretMarkerType : signed char {
+      NONE,
+      CIRCLE,
+      CROSS
+      };
+
+
+class FretItem {
+   public:
+      struct Barre {
+            int startString;
+            int endString;
+
+            Barre() { startString = endString = -1; }
+            Barre(int s, int e) : startString(s), endString(e) {}
+            bool exists() const { return startString > -1; }
+            };
+
+      struct Dot {
+            int fret                { 0 };
+            FretDotType dtype       { FretDotType::NORMAL };
+            int fingering;          // NOTE:JT - possible future feature?
```

Comment:
I was going to add fingering but didn't get round to it. I plan on doing it in a future PR, so rather than removing all the traces of fingering code, I decided to just leave it in for now. It doesn't have any negative effects, AFAIK...

But, if you're sure, I can remove it if you think that'd be better.